tycho-packaging-plugin/src/main/java/org/eclipse/tycho/packaging/PackageFeatureMojo.java

```diff
@@ -168,6 +191,18 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             jarArchiver.setDestFile(outputJar);
 
             try {
+                // Additional filesets win over bin.includes ones, so we add them first
+                if (additionalFileSets != null) {
+                    for (final var fileSet : additionalFileSets) {
+                        final var directory = fileSet.getDirectory();
+
+                        // noinspection ConstantConditions
+                        if (directory != null && directory.isDirectory()) {
```

Comment:
I'm not sure why these checks are present in `package-plugin`. There is a `@Nonnull` annotation, and the method explicitly states it's a directory. Can we get rid of them?